Data/Aeson/Types/Instances.hs

```diff
@@ -1693,11 +1693,16 @@ contramapToJSONKeyFunction h x = case x of
     ToJSONKeyText (f,g) -> ToJSONKeyText (f . h, g . h)
     ToJSONKeyValue (f,g) -> ToJSONKeyValue (f . h, g . h)
 
-mapFromJSONKeyFunction :: (a -> b) -> FromJSONKeyFunction a -> FromJSONKeyFunction b
+-- I'm not sure if the 'FromJSONKeyCoerce' branch makes this not be
+-- a valid fmap implementation.
+mapFromJSONKeyFunction :: forall a b. (a -> b) -> FromJSONKeyFunction a -> FromJSONKeyFunction b
 mapFromJSONKeyFunction h x = case x of
+    FromJSONKeyCoerce -> FromJSONKeyText (h . uc)
```

Comment:
Ah good spot. Should we remove this. The definition of `fmap` in `Functor` won't work otherwise. (it works for `Tagged`,
because it's a newtype itself!)